src/api/app/models/workflow/step.rb

```diff
@@ -111,6 +111,20 @@ def create_subscriptions(package, workflow_filters)
     end
   end
 
+  def destroy_subscriptions(package, _workflow_filters)
+    ['Event::BuildFail', 'Event::BuildSuccess'].each do |build_event|
+      subscription = EventSubscription.find_by(eventtype: build_event,
+                                               channel: 'scm',
+                                               token: @token,
+                                               package: package)
+      subscription.destroy
```

Comment:
There are no before/after destroy callbacks on the `EventSubscription` model, so we could delete them all in one SQL query with `EventSubscription.where(...).delete_all`. This would be more efficient.

Reply:
so, I just uncommented my code and adapted to use `delete_all` @dmarcoux